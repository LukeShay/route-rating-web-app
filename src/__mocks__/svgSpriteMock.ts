export default {
  id: 'svg-sprite-id',
  viewBox: '0 0 1 1',
  url: '/sprite.svg#svg-sprite-id',
};
