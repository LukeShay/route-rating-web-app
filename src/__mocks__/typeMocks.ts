import { Gym, Route, User, Wall } from "../types";

export const testRouteOne: Route = {
  id: "route1",
  wallId: "wall1",
  gymId: "gym1",
  name: "RouteOne",
  setter: "Luke",
  holdColor: "Blue",
  types: ["TOP_ROPE", "LEAD"],
  averageGrade: "GRADE_5_9",
  averageRating: 4
};

export const testRouteTwo: Route = {
  id: "route2",
  wallId: "wall1",
  gymId: "gym1",
  name: "RouteTwo",
  setter: "Luke",
  holdColor: "Blue",
  types: ["TOP_ROPE", "LEAD"],
  averageGrade: "GRADE_5_12ab",
  averageRating: 4
};

export const testWallOne: Wall = {
  id: "wall1",
  gymId: "gym1",
  name: "Wall1",
  routes: [testRouteOne, testRouteTwo],
  types: ["LEAD", "TOP_ROPE"]
};

export const testWallTwo: Wall = {
  id: "wall2",
  gymId: "gym1",
  name: "Wall1",
  routes: [],
  types: ["LEAD", "TOP_ROPE"]
};

export const testGymOne: Gym = {
  address: "1234 Five Street",
  authorizedEditors: ["id"],
  city: "Ames",
  email: "abc@d.com",
  id: "gym1",
  logoUrl: "",
  name: "Gym One",
  phoneNumber: "1234567890",
  photoUrl: "",
  state: "Iowa",
  walls: [testWallOne, testWallTwo],
  website: "lukeshay.com",
  zipCode: "50014"
};

export const testGymTwo: Gym = {
  address: "1234 Five Street",
  authorizedEditors: ["id"],
  city: "Ames",
  email: "abc@d.com",
  id: "gym2",
  logoUrl: "",
  name: "Gym Two",
  phoneNumber: "1234567890",
  photoUrl: "",
  state: "Iowa",
  walls: [testWallOne, testWallTwo],
  website: "lukeshay.com",
  zipCode: "50014"
};

export const testUser: User = {
  username: "username",
  email: "email",
  password: "password",
  id: "id",
  authority: "ADMIN",
  country: "USA",
  state: "IA",
  firstName: "Name",
  lastName: "Last",
  phoneNumber: "1111111111",
  role: "ADMIN_ROLE",
  session: null
};
