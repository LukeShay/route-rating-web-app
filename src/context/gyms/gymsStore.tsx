import React, { createContext, Dispatch, Reducer, useReducer } from "react";
import { Gym } from "../../types";
import Types from "./gymsActionTypes";

export interface IGymsContextState {
  gyms: Gym[];
}

export interface IGymsContextAction {
  actionType: string;
  gyms: Gym[];
  gym?: Gym;
}

export interface IContextProps {
  state: IGymsContextState;
  dispatch: Dispatch<IGymsContextAction>;
}

export const GymsContext = createContext<IContextProps>({} as IContextProps);

const reducer: Reducer<IGymsContextState, IGymsContextAction> = (
  state: IGymsContextState,
  { actionType, gyms, gym }: IGymsContextAction
): IGymsContextState => {
  switch (actionType) {
    case Types.LOAD_GYMS:
      return { gyms };

    case Types.UPDATE_GYM:
      if (!gym) {
        throw new Error("Action must have a gym.");
      } else {
        return {
          gyms: state.gyms.map((gymFromList: Gym) =>
            gym.id === gymFromList.id ? gym : gymFromList
          )
        };
      }

    default:
      throw new Error("Action type must be defined");
  }
};

const initialState: IGymsContextState = {
  gyms: Array<Gym>()
};

export const GymsStore: React.FC = ({ children }): JSX.Element => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <GymsContext.Provider value={{ state, dispatch }}>
      {children}
    </GymsContext.Provider>
  );
};

export const useGymsContext = (): IContextProps =>
  React.useContext(GymsContext);
