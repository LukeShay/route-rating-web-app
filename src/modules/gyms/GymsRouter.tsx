import React from "react";
import { Route, Switch } from "react-router";
import { AuthRoutes, Routes } from "../../routes";

const GymsPage = React.lazy(() => import("./views/GymsPage"));
const GymPage = React.lazy(() => import("./views/GymPage"));
const GymEditPage = React.lazy(() => import("./views/GymEditPage"));

const GymRouter: React.FC = (): JSX.Element => (
  <React.Suspense fallback={<div />}>
    <Switch>
      <Route exact={true} path={Routes.GYMS} component={GymsPage} />
      <Route path={AuthRoutes.EDIT_GYM} component={GymEditPage} />
      <Route path={Routes.GYMS} component={GymPage} />
    </Switch>
  </React.Suspense>
);

export default GymRouter;
