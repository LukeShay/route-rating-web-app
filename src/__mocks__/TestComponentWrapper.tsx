/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IViewContextState } from "../context/view/viewStore";
import { IUserContextState } from "../context/user/userStore";
import { StaticRouter } from "react-router";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperProps {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  location: string;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapper: React.FC<ITestComponentWrapperProps> = ({
  children,
  gymsState,
  location,
  userState,
  viewState
}): JSX.Element => {
  return (
    <StaticRouter location={`https://lukeshay.com/${location}`}>
      <GymsStoreMock state={gymsState} dispatch={() => {}}>
        <ViewStoreMock state={viewState} dispatch={() => {}}>
          <UserStoreMock state={userState} dispatch={() => {}}>
            {children}
          </UserStoreMock>
        </ViewStoreMock>
      </GymsStoreMock>
    </StaticRouter>
  );
};

export default TestComponentWrapper;
