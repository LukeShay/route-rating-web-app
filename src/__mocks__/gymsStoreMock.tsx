import React from "react";
import { IGymsContextState, GymsContext } from "../context/gyms/gymsStore";

export interface IGymsStoreMockProps {
  state: IGymsContextState;
  dispatch: any;
}

const GymsStoreMock: React.FC<IGymsStoreMockProps> = ({
  state,
  dispatch,
  children
}): JSX.Element => (
  <GymsContext.Provider value={{ state, dispatch }}>
    {children}
  </GymsContext.Provider>
);

export default GymsStoreMock;
