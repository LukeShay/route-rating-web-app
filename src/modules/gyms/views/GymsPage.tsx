import { createStyles, makeStyles } from '@material-ui/core';
import React from 'react';
import { useHistory } from 'react-router-dom';
import { toast } from 'react-toastify';
import * as GymsActions from '../../../context/gyms/gymsActions';
import { useGymsContext } from '../../../context/gyms/gymsStore';
import { Routes } from '../../../routes';
import { Events } from '../../../types';
import Input from '../../common/inputs/Input';
import { useViewContext } from '../../../context/view/viewStore';
import Tabs from '@material-ui/core/Tabs';
import Paper from '@material-ui/core/Paper';
import Tab from '@material-ui/core/Tab';
import GymsList from './components/GymsList';

const useStyles = makeStyles(() =>
  createStyles({
    div: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
      width: '100%',
    },
    card: {
      borderRadius: '10px',
      height: '233px',
      width: '700px',
      display: 'flex',
    },
    cardWrapper: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
      paddingBottom: '10px',
      paddingTop: '10px',
      width: '100%',
    },
    information: {
      paddingBottom: '5px',
      paddingLeft: '10px',
      paddingTop: '5px',
    },
    photo: {
      borderRadius: '10px',
      height: '96%',
    },
    photoWrapper: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
      width: '50%',
    },
    root: {
      width: '100%',
    },
    search: {
      width: '50%',
    },
    searchMobile: {
      width: '96%',
    },
  })
);

const GymsPage: React.FC = (): JSX.Element => {
  const { state: gymsState, dispatch: gymsDispatch } = useGymsContext();
  const { state: viewState } = useViewContext();

  const history = useHistory();

  const [search, setSearch] = React.useState<string>('');
  const [page, setPage] = React.useState<number>(0);

  const classes = useStyles();

  const loadGyms = (query, page): void => {
    GymsActions.loadGymsQuery(gymsDispatch, query, page).then((response) => {
      if (!response || !(response instanceof Response) || !response.ok) {
        toast.error('Error getting gyms.');
      }
    });
  };

  React.useEffect(() => {
    if (
      !gymsState.page ||
      !gymsState.page.content ||
      gymsState.page.content.length === 0
    ) {
      loadGyms('', page);
    }
  }, []);

  const searchClass = (): string => {
    return viewState.mobile ? classes.searchMobile : classes.search;
  };

  const handleSearchChange = (event: Events.InputEvent): void => {
    const { value } = event.target;
    setSearch(value);
  };

  const handlePageChange = (
    event: React.ChangeEvent<{}>,
    newValue: number
  ): void => {
    const newPage = page + newValue - 1;
    if (newPage >= 0 && newPage < gymsState.page.totalPages) {
      setPage(newPage);
      loadGyms(search, page + newValue - 1);
    }
  };

  const handleKeyPress = (event: Events.KeyboardEvent): void => {
    if (event.key === 'Enter') {
      loadGyms(search, page);
    }
  };

  const handleRowPress = (id: string): void =>
    history.push(Routes.GYMS + '/' + id);

  return (
    <div className={classes.root} data-test-id="gym-page-test-id">
      <div className={classes.div}>
        <Input
          className={searchClass()}
          fullWidth={false}
          id="search"
          name="search"
          onChange={handleSearchChange}
          onKeyPress={handleKeyPress}
          placeholder="Search"
          type="text"
          value={search}
        />
      </div>
      <GymsList
        cardClass={classes.card}
        gyms={gymsState.page.content || []}
        mobile={viewState.mobile}
        onClick={handleRowPress}
      />
      <Paper>
        <Tabs
          value={1}
          id="page"
          onChange={handlePageChange}
          indicatorColor="primary"
          textColor="primary"
          centered
        >
          <Tab label="<" />
          <Tab label={page + 1} />
          <Tab label=">" />
        </Tabs>
      </Paper>
    </div>
  );
};

export default React.memo(GymsPage);
