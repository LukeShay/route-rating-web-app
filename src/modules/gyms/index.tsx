import React from "react";
import GymsRouter from "./GymsRouter";

const GymsIndex: React.FC = (): JSX.Element => <GymsRouter />;

export default GymsIndex;
