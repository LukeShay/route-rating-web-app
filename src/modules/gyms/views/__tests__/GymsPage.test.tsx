import React from "react";
import GymsPage from "../GymsPage";
import GymsStoreMock from "../../../../__mocks__/gymsStoreMock";
import { IGymsContextState } from "../../../../context/gyms/gymsStore";
import * as TypeMocks from "../../../../__mocks__/typeMocks";
import { shallow } from "../../../../../configs/setupEnzyme";
import { shallowToJson } from "enzyme-to-json";

const mockGymsState: IGymsContextState = {
  gyms: [TypeMocks.testGymOne, TypeMocks.testGymTwo]
};

describe("<GymsPage />", () => {
  it("should render correctly", () => {
    const gymsPage = shallow(
      <GymsStoreMock state={mockGymsState} dispatch={() => {}}>
        <GymsPage />
      </GymsStoreMock>
    );

    expect(shallowToJson(gymsPage)).toMatchSnapshot();



  });
});
