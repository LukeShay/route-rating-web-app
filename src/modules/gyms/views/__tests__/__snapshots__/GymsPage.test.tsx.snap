// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GymsPage /> should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "gyms": Array [
          Object {
            "address": "1234 Five Street",
            "authorizedEditors": Array [
              "id",
            ],
            "city": "Ames",
            "email": "abc@d.com",
            "id": "gym1",
            "logoUrl": "",
            "name": "Gym One",
            "phoneNumber": "1234567890",
            "photoUrl": "",
            "state": "Iowa",
            "walls": Array [
              Object {
                "gymId": "gym1",
                "id": "wall1",
                "name": "Wall1",
                "routes": Array [
                  Object {
                    "averageGrade": "GRADE_5_9",
                    "averageRating": 4,
                    "gymId": "gym1",
                    "holdColor": "Blue",
                    "id": "route1",
                    "name": "RouteOne",
                    "setter": "Luke",
                    "types": Array [
                      "TOP_ROPE",
                      "LEAD",
                    ],
                    "wallId": "wall1",
                  },
                  Object {
                    "averageGrade": "GRADE_5_12ab",
                    "averageRating": 4,
                    "gymId": "gym1",
                    "holdColor": "Blue",
                    "id": "route2",
                    "name": "RouteTwo",
                    "setter": "Luke",
                    "types": Array [
                      "TOP_ROPE",
                      "LEAD",
                    ],
                    "wallId": "wall1",
                  },
                ],
                "types": Array [
                  "LEAD",
                  "TOP_ROPE",
                ],
              },
              Object {
                "gymId": "gym1",
                "id": "wall2",
                "name": "Wall1",
                "routes": Array [],
                "types": Array [
                  "LEAD",
                  "TOP_ROPE",
                ],
              },
            ],
            "website": "lukeshay.com",
            "zipCode": "50014",
          },
          Object {
            "address": "1234 Five Street",
            "authorizedEditors": Array [
              "id",
            ],
            "city": "Ames",
            "email": "abc@d.com",
            "id": "gym2",
            "logoUrl": "",
            "name": "Gym Two",
            "phoneNumber": "1234567890",
            "photoUrl": "",
            "state": "Iowa",
            "walls": Array [
              Object {
                "gymId": "gym1",
                "id": "wall1",
                "name": "Wall1",
                "routes": Array [
                  Object {
                    "averageGrade": "GRADE_5_9",
                    "averageRating": 4,
                    "gymId": "gym1",
                    "holdColor": "Blue",
                    "id": "route1",
                    "name": "RouteOne",
                    "setter": "Luke",
                    "types": Array [
                      "TOP_ROPE",
                      "LEAD",
                    ],
                    "wallId": "wall1",
                  },
                  Object {
                    "averageGrade": "GRADE_5_12ab",
                    "averageRating": 4,
                    "gymId": "gym1",
                    "holdColor": "Blue",
                    "id": "route2",
                    "name": "RouteTwo",
                    "setter": "Luke",
                    "types": Array [
                      "TOP_ROPE",
                      "LEAD",
                    ],
                    "wallId": "wall1",
                  },
                ],
                "types": Array [
                  "LEAD",
                  "TOP_ROPE",
                ],
              },
              Object {
                "gymId": "gym1",
                "id": "wall2",
                "name": "Wall1",
                "routes": Array [],
                "types": Array [
                  "LEAD",
                  "TOP_ROPE",
                ],
              },
            ],
            "website": "lukeshay.com",
            "zipCode": "50014",
          },
        ],
      },
    }
  }
>
  <Memo(GymsPage) />
</ContextProvider>
`;
